core/citrus-base/src/main/java/com/consol/citrus/container/WaitConditionBuilder.java

```diff
@@ -25,26 +25,20 @@
  * @author Christoph Deppisch
  * @since 2.4
  */
-public abstract class WaitConditionBuilder<T extends Condition, S extends WaitConditionBuilder> implements TestActionBuilder<Wait> {
+public abstract class WaitConditionBuilder<T extends Condition, S extends WaitConditionBuilder<T, S>> implements TestActionBuilder<Wait> {
 
     /** Parent wait action builder */
-    private final Wait.Builder builder;
-
-    /** Condition */
-    private final T condition;
+    private final Wait.Builder<T> builder;
 
     /** Self reference */
     protected S self;
 
     /**
      * Default constructor using fields.
-     * @param condition
      * @param builder
      */
-    public WaitConditionBuilder(T condition, Wait.Builder builder) {
-        this.condition = condition;
+    public WaitConditionBuilder(Wait.Builder<T> builder) {
```

Comment:
the problem is, this change is potentially breaking @christophd.. should we target it for citrus 4.0? i am working on that, but have a silly error with `citrus-docker` and don't know what's up there.

Reply:
yes, it is never backwards compatible as soon as you just remove an argument from a constructor. if someone uses this constructor, he's doomed :(

Reply:
@christophd actually, I was just able to resolve the `citrus-docker` compilation failure.. maybe I can manage to provide a fullbuild of citrus@v4.0.0 until this evening.

Reply:
@bbortt to be honest I think it would be ok to change the constructor even on a minor version upgrade. I know people may need to adjust their code but the change is trivial and the API change is done on a kind of internal API. People using the end user Java DSL will not even notice when updating. 

I know most of the libraries are doing it this way when the changes are trivial and when there is a notice of the change added to the release notes (that we do not have at the moment which is a problem).

Summing up I would add this for upcoming version 3.3.0 if we agree to have a decent release notes section.

Reply:
I am fine with that. did also note it in #878.